Patronus Evaluate API Integration

varjoshi opened the pull request:
## Description

Patronus AI manages a powerful suite of fully-managed in-house evaluation models. They can be easily access through the Patronus Evaluate API. This PR adds an integration with this API so it can be used as an output rail.

cc @drazvan 

## Related Issue(s)

N/A
## Checklist

- [✔️ ] I've read the [CONTRIBUTING](https://github.com/NVIDIA/NeMo-Guardrails/blob/develop/CONTRIBUTING.md) guidelines.
- [✔️ ] I've updated the documentation if applicable.
- [ ✔️] I've added tests if applicable.
- [ ✔️] @mentions of the person or team responsible for reviewing proposed changes.

Pouyanpi reviewed (commented):
@varjoshi Thanks for your PR! It looks good, just review the suggestions and I also noticed that some functions are not tested. Once you have done the requested changes we are good to merge. Thanks!

Functions missing test:
actions.py:
`check_guardrail_pass`
`parse_patronus_lynx_response`

varjoshi commented on nemoguardrails/library/patronusai/actions.py:
```diff
@@ -108,3 +110,136 @@ async def patronus_lynx_check_output_hallucination(
+                    "The Patronus Evaluate API call failed with status code %s. Details: %s",
+                    response.status,
+                    await response.text(),
+                )
```
Sure! Will dd more to the docstring

varjoshi commented:
> @varjoshi Thanks for your PR! It looks good, just review the suggestions and I also noticed that some functions are not tested. Once you have done the requested changes we are good to merge. Thanks!
> 
> Functions missing test: actions.py: `check_guardrail_pass` `parse_patronus_lynx_response`

Hi @Pouyanpi - thanks for your review. I've addressed your comments and added the tests you requested.

Pouyanpi commented on tests/test_patronus_evaluate_api.py:
```diff
@@ -0,0 +1,983 @@
+                tags: { "hello": "world" },
+              }
+    """
+    print("rails config", YAML_PREFIX + yaml_evaluate_config)
```
remove print

Pouyanpi replied:
```suggestion
```

Pouyanpi commented on tests/test_patronus_evaluate_api.py:
```diff
@@ -0,0 +1,983 @@
+                tags: { "hello": "world" },
+              }
+    """
+    print("rails config", YAML_PREFIX + yaml_evaluate_config)
```
```suggestion
```

Pouyanpi commented on tests/test_patronus_evaluate_api.py:
```diff
@@ -0,0 +1,983 @@
+                tags: { "hello": "world" },
+              }
+    """
+    print("rails config", YAML_PREFIX + yaml_evaluate_config)
```
```suggestion
```

Pouyanpi commented on tests/test_patronus_evaluate_api.py:
```diff
@@ -0,0 +1,983 @@
+                tags: { "hello": "world" },
+              }
+    """
+    print("rails config", YAML_PREFIX + yaml_evaluate_config)
```
```suggestion
```

Pouyanpi reviewed (commented):
@varjoshi Thank you for adding the tests. It looks good 👍🏻
